src/csharp/Grpc.Core/Utils/AsyncStreamExtensions.cs

```diff
@@ -27,12 +28,41 @@ namespace Grpc.Core.Utils
     /// </summary>
     public static class AsyncStreamExtensions
     {
+        /// <summary>
+        /// Advances the stream reader to the next element in the sequence, returning the result asynchronously.
+        /// </summary>
+        /// <typeparam name="T">The message type.</typeparam>
+        /// <param name="streamReader">The stream reader.</param>
+        /// <returns>
+        /// Task containing the result of the operation: true if the reader was successfully advanced
+        /// to the next element; false if the reader has passed the end of the sequence.
+        /// </returns>
+        public static Task<bool> MoveNext<T>(this IAsyncStreamReader<T> streamReader)
```

Comment:
putting this extension method under  Grpc.Core.Utils namespace means that for everyone who's using MoveNext() (originally under a different namespace) and is not importing Grpc.Core.Utils, the compilation will be broken. It seems safer to put the extension method under Grpc.Core namespace to avoid this?

Reply:
It's a breaking change either way, but you're right that having it located in Grpc.Core means less source code needs to be updated.